ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/ExecutionBuilderModule.java

```diff
@@ -518,16 +543,44 @@ private void logReceivedBuilderBid(final BuilderBid builderBid) {
         blobsLog);
   }
 
-  private void logLocalPayloadWin(final UInt256 builderBidValue, final UInt256 localPayloadValue) {
-    if (builderBidCompareFactor.isEmpty() || builderBidCompareFactor.get() == HUNDRED_PERCENT) {
+  private void logLocalPayloadWin(
+      final UInt256 builderBidValue,
+      final UInt256 localPayloadValue,
+      final Optional<UInt64> requestedBuilderBoostFactor) {
+    final boolean isDefaultComparison;
+    final Optional<?> actualComparisonFactor;
+    final String comparisonFactorSource;
+
+    if (requestedBuilderBoostFactor.isPresent()) {
+      // If the requestedBuilderBoostFactor is set,
+      // we always use it over builderBidCompareFactor to determine whether the local payload wins
+      isDefaultComparison =
```

Comment:
Not a claim, just a suggestion: I think max uint64% will look ugly in log. We could probably use some string keywords for 0 and max, like you did already with `MAX_PROFIT`

Reply:
ah you're right. I'll do that